op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,138 @@
+	require.False(t, is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain B")
+
+	// Submit all and mine for both chains
+	actors.ChainA.Sequencer.ActL2EmptyBlock(t)
+	actors.ChainA.Batcher.ActSubmitAll(t)
+	actors.ChainB.Sequencer.ActL2EmptyBlock(t)
+	actors.ChainB.Batcher.ActSubmitAll(t)
+	batchTxA := actors.ChainA.Batcher.LastSubmitted
+	batchTxB := actors.ChainB.Batcher.LastSubmitted
+	actors.L1Miner.ActL1StartBlock(12)(t)
+	actors.L1Miner.ActL1IncludeTxByHash(batchTxA.Hash())(t)
+	actors.L1Miner.ActL1IncludeTxByHash(batchTxB.Hash())(t)
+	actors.L1Miner.ActL1EndBlock(t)
+
+	rollupConfigA := is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg
+	rollupConfigB := is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg
+
+	currentBlockHeaderA := actors.ChainA.SequencerEngine.L2Chain().CurrentBlock()
+	currentBlockHeaderB := actors.ChainB.SequencerEngine.L2Chain().CurrentBlock()
+	require.False(t, rollupConfigA.IsInterop(currentBlockHeaderA.Time),
+		"Interop should not be active at the first L1 inclusion block in chain A")
+	require.False(t, rollupConfigB.IsInterop(currentBlockHeaderB.Time),
+		"Interop should not be active at the first L1 inclusion block in chain B")
+
+	// Build a few L2 blocks. We only need the L1 inclusion to advance past Interop and Interop
+	// shouldn't activate with L2 time.
+	actors.ChainA.Sequencer.ActBuildL2ToInterop(t)
+	actors.ChainB.Sequencer.ActBuildL2ToInterop(t)
+
+	////////////////////////////
+	// Post-upgrade Block Production
+	////////////////////////////
+
+	currentBlockHeaderA = actors.ChainA.SequencerEngine.L2Chain().CurrentBlock()
+	currentBlockHeaderB = actors.ChainB.SequencerEngine.L2Chain().CurrentBlock()
+	require.Equal(t, currentBlockHeaderA.Number.Uint64(), currentBlockHeaderB.Number.Uint64())
+	currentBlockNumber := currentBlockHeaderA.Number
+
+	require.True(t, rollupConfigA.IsInteropActivationBlock(currentBlockHeaderA.Time),
+		"Interop should be active at the first L1 inclusion block in chain A")
+	require.True(t, rollupConfigB.IsInteropActivationBlock(currentBlockHeaderB.Time),
+		"Interop should be active at the first L1 inclusion block in chain B")
+
+	currentBlock := actors.ChainA.SequencerEngine.L2Chain().GetBlockByHash(currentBlockHeaderA.Hash())
+	activationBlockTxs := currentBlock.Transactions()
+
+	require.Len(t, activationBlockTxs, 5) // 4 upgrade txs + 1 system deposit tx
+	upgradeTransactions := activationBlockTxs[1:]
+	upgradeTransactionBytes := make([]hexutil.Bytes, len(upgradeTransactions))
+	for i, tx := range upgradeTransactions {
+		txBytes, err := tx.MarshalBinary()
+		require.NoError(t, err)
+		upgradeTransactionBytes[i] = txBytes
+	}
+
+	expectedUpgradeTransactions, err := derive.InteropNetworkUpgradeTransactions()
+	require.NoError(t, err)
+
+	require.ElementsMatch(t, upgradeTransactionBytes, expectedUpgradeTransactions)
```

Comment:
Why can we only match the elements? Shouldn't the order also be preserved?

Reply:
Good point, I was looking for something arraylike because I hadn't discovered that testify/require.Equal() works for element-wise ordering-sensitive comparison for slices. Fixed in c6fd3b743583a4bdbc626c27c8e5b1a8af210d24.